### PR TITLE
2016 q4.fix.odplayerh264

### PR DESCRIPTION
**Pull request by chrberger**

### CODE REVIEW

**Comment by chrberger** [comment]
@HangYin2015 May I ask you to review this PR?

File: odcockpit/src/plugins/player/PlayerWidget.cpp [hunk 1]
```diff
@@ -189,43 +191,45 @@ namespace cockpit {
...
-                    // Increment the counters.
                     if (m_player->hasMoreData()) {
-                        m_containerCounter++;
-                        if (!(m_containerCounter < m_containerCounterTotal)) {
-                            m_containerCounterTotal = m_containerCounter;
+                        // Get container to be sent.
+                        Container nextContainerToBeSent = m_player->getNextContainerToBeSent();
+
+                        // Increment the counters.
+                        if (m_player->hasMoreData()) {
```
Comment (HangYin2015) [inline odcockpit/src/plugins/player/PlayerWidget.cpp:204]: This if condition is already present in the parent if branch two lines above. Do we also need it here?

Comment (chrberger) [reply odcockpit/src/plugins/player/PlayerWidget.cpp:204]: getNextContainer...() might modify the hasNextContainer state. That's the reason why we need the if-statement here.

Comment (HangYin2015) [reply odcockpit/src/plugins/player/PlayerWidget.cpp:204]: OK. Thanks. Then it makes sense.

**Review by HangYin2015 — APPROVED** [review]

**Pull request merged**